test/integration/test_utils/git.py

```diff
@@ -0,0 +1,122 @@
+
+    commands = [
+        f'git -c init.defaultBranch=master init . {" ".join(git_init_args)}'
+    ]
+
+    command = ' '.join(commands)
+    check_output(command, cwd=str(path), shell=True)
+
+    if bare:
+        # NOTE: You cannot ``git clone file:///location/to/bare/repo -b master``, even
+        #       if the repository exists and has the master branch configured. First,
+        #       you need to push something so Git will create the refs/master for the
+        #       branch. We do it here, since Autosubmit uses ``-b`` by default.
+        with TemporaryDirectory() as td:
+            clone_repo = Path(td, 'clone')
+            git_clone_repository(f'file:///{str(path)}', clone_repo)
+
+            with open(clone_repo / 'README.md', 'w') as f:
+                f.write('This is a test repository of Autosubmit.')
+
+            git_commit_all_in_dir(clone_repo, branch=branch, push=True)
+
+
+def git_clone_repository(url: str, path: Path) -> None:
+    """Clone a Git repository into the given path."""
+    if not path or not path.is_absolute() or path.exists():
+        raise ValueError(f'You must provide a valid, absolute, non-existent path for your Git repository: {path}')
+
+    if not url:
+        raise ValueError(f'You must provide a valid URL to be cloned: {url}')
+
+    commands = [
+        f'git clone {url} {str(path)}'
+    ]
+
+    command = ' '.join(commands)
+    # Change cwd even if the path is absolute, just in case...
+    check_output(command, cwd=str(path.parent), shell=True)
+
+
+def git_commit_all_in_dir(path: Path, push=False, remote='origin', branch='master') -> None:
+    """Adds all files in the given path, and creates a single commit for it."""
+    if not path or not path.is_dir():
+        raise ValueError(f'You must provide a valid path for your Git repository: {path}')
+
+    if not list(path.iterdir()):
+        raise ValueError(f'Tried to commit all files, but directory is empty: {path}')
+
+    commands = [
+        'git add .',
+        'git commit -am "Initial commit"'
+    ]
+
+    if push:
+        commands.append(f'git push {remote} {branch}')
+
+    command = ';'.join(commands)
+    check_output(command, cwd=str(path), shell=True)
+
+
+def git_add_submodule(url: str, path: Path, name: str, push=False, remote='origin', branch='master') -> None:
+    """Adds a submodule to a Git repository."""
+    if not path or not path.is_dir():
+        raise ValueError(f'You must provide a valid path for your Git repository: {path}')
+
+    if not url:
+        raise ValueError(f'You must provide a valid URL to be cloned: {url}')
+
+    commands = [
+        'git submodule init',
+        f'git -c protocol.file.allow=always submodule add {url} {name}',
+        'git add .',
+        'git commit -am "Add submodule"'
+    ]
+
+    if push:
+        commands.append(f'git push {remote} {branch}')
+
+    command = ';'.join(commands)
+    check_output(command, cwd=str(path), shell=True)
```

Comment:
These are handy functions for writing tests that manipulate contents of Git repositories. In the future, we can use these to test the remaining `git` package -- which has only 45.64% of coverage right now, but shouldn't be too hard to test the rest.

<img width="1248" height="563" alt="image" src="https://github.com/user-attachments/assets/6a38c46a-f63b-4d6c-ab92-a4148d9e85d4" />